src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSnapshotManagerDispatcherBase.cs

```diff
@@ -66,11 +71,17 @@ private void ThreadStart()
                     try
                     {
                         var task = _tasks.Take();
-                        TryExecuteTask(task);
+                        if (!TryExecuteTask(task))
+                        {
+                            Debug.WriteLine("Task failed to execute on thread " + _threadName + ".");
```

Comment:
Ah, instead of `Debug.WriteLine` we should take in our [ErrorReporter](https://github.com/dotnet/aspnetcore-tooling/blob/d08ac07ef8dda3b7df7dc61bd55ee25577ca3a73/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ErrorReporter.cs) contract and log errors. That way the ActivityLog or LogHub (depends on where things explode, LSP versus VS client) will have appropriate logs to allow us to investigate